### DISCUSSION TITLE
Upgrade from 6.15.2 to 7.10 - need advise/help

### DISCUSSION THREAD

**Discussion by s0p4LiN** [Keyfactor/ejbca-ce#181] [Q&A] [answered]
Hello, I could not find any information on google so I come here.

In may 2020, I had installed EJBCA 6.15.2.6 Community (r34564) for the company I work for. 
I was in apprenticeship at the time and did not wrote any documentation. I've just followed at the time followed these guides: https://fladnag.net/?page_id=104 as the PrimeKey documentation was too difficult to understand.

But now, the Certificate Authority is used by all of our Internal Web Services and I want to upgrade it as it been 2 years in production already.

EJBCA were installed on an Debian 10 with:
- Wildfly 10.1.0
- openjdk-8-jdk
- mariadb  Ver 15.1 Distrib 10.5.9-MariaDB

I want to upgrade EJBCA first, and then migrate it to an Ubuntu 22.04 LTS Server.

Here's the files in my ejbca-custom folder:

```
ejbca.properties
database.properties
crlstore.properties
cesecore.properties
certstore.properties
web.properties
install.properties
batchtool.properties
mail.properties
```

I tried to follow the official documentation for upgrading the production server, but if I'm stuck at merging *.properties files, I do not understand the terms.

Do I need to add all new line from the *.properties files that comes from the 7.10 versions into my current *.properties files and then continue the upgrade with `ant deployear` ?

Also, do I need to upgrade OpenJDK and Wildfly after EJBCA upgrade ?

And last question, for the databases script update, which one do I need to use ?
If im logical, I would say the scripts in 615_70, but as I see folder 790_7100, it means the scripts in the folder 615_70 helps upgrading from v6.15 to v7.0 but not to v7.10 ?

Thank you

**Comment by mike-agrenius-kushner** [comment, marked as answer]
Hi, 

I'd say https://doc.primekey.com/ejbca/ejbca-installation/upgrading-ejbca is fairly straight forward, but to be clear:

1. You don't need to touch the property files at all. The beauty of having them in ejbca-custom means that 7.10 will just source them from there when constructing the EAR
2. If your wildfly user has create rights to your database, you likewise need to do nothing – EJBCA will create all the new tables/columns for you and update any needed data. Afterwards you'll need to run the post-upgrade from the UI – this removes old data from the database and exists to allow multiple EJBCA versions in a cluster to run against the same database during upgrade. 

So all you need to do is run `ant deployear` from the main directory of EJBCA 7.10, which needs to be placed in the same directory as ejbca-custom. 

Cheers,
Mike

**Reply by s0p4LiN** [reply]
Thank you for the answer.

I did the upgrade now I understand how it works.  But it look like the post upgrade is stuck:

![image](https://user-images.githubusercontent.com/68989865/205319056-fcc7f3fa-8758-442e-80b1-da99c3bf4447.png)

It stopped at Upgrade to 7.4.0 and now nothing happens.

Do you have any recommandation ?

**Reply by mike-agrenius-kushner** [reply]
Hi, 

Huh, now that is odd. Do you still have the option to run post-upgrade? 

What do they log files say? 

Cheers,
Mike

**Reply by s0p4LiN** [reply]
```
<html>
<body>
<!--StartFragment--><div class="container" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; color: rgb(0, 0, 0); font-size: 14.6667px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div id="messagesAndContent" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; min-height: 100%; padding-left: 30px; padding-right: 15px;"><div id="contentBlock" class="contentBlock" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; padding-top: 7px; padding-bottom: 40px;"><form id="upgrade" name="upgrade" method="post" action="https://root-ca.superprod.lan:8443/ejbca/adminweb/upgrade.xhtml" class="fullwidth readablewidth" enctype="application/x-www-form-urlencoded" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; max-width: 1000px; width: 1000px;"><div id="upgrade:log" class="" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif;">

2022-12-02 15:37:42+01:00 | INFO | Database content version: 6.10.1, current application version: 7.10.0.2 -> Starting post-upgrade.
-- | -- | --
2022-12-02 15:37:42+01:00 | INFO | Starting post upgrade to 7.2.0
2022-12-02 15:37:42+01:00 | INFO | Post upgrade to 7.2.0 complete.
2022-12-02 15:37:42+01:00 | INFO | Starting post upgrade to 7.4.0
2022-12-02 15:37:42+01:00 | INFO | Successfully normalized 73 rows in CRLData. Completed in 101 ms.
2022-12-02 16:37:42+01:00 | INFO | Successfully updated index 'crldata_idx3' for database table 'CRLData'. Completed in 3599912 ms.
2022-12-02 16:37:42+01:00 | INFO | Successfully updated index 'crldata_idx4' for database table 'CRLData'. Completed in 33 ms.
2022-12-02 16:37:42+01:00 | INFO | Post upgrade to 7.4.0 complete.
2022-12-02 16:37:42+01:00 | ERROR | Error thrown during upgrade: <- org.hibernate.exception.GenericJDBCException: Unable to acquire JDBC Connection <- org.hibernate.exception.GenericJDBCException: Unable to acquire JDBC Connection <- Unable to acquire JDBC Connection <- javax.resource.ResourceException: IJ000460: Error checking for a transaction <- IJ000460: Error checking for a transaction <- IJ000459: Transaction is not active: tx=TransactionImple < ac, BasicAction: 0:ffff7f000001:-58bdd4ca:638a0aec:1a8 status: ActionStatus.ABORTED >
2022-12-02 16:37:42+01:00 | ERROR | Unexpected error from post-upgrade: Transaction rolled back <- Transaction rolled back <- WFLYEJB0447: Transaction 'TransactionImple < ac, BasicAction: 0:ffff7f000001:-58bdd4ca:638a0aec:1a8 status: ActionStatus.ABORTED >' was already rolled back

</div></form></div></div></div><div id="footerBlock" class="footerBlock" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; position: relative; margin-top: -47px; height: 40px; padding-left: 30px; padding-right: 15px; clear: both; text-align: center; color: rgb(0, 0, 0); font-size: 14.6667px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div id="footerInner" class="footerInner" style="font-family: Verdana, Arial, Helvetica, Geneva, sans-serif; line-height: 40px; vertical-align: middle; height: 40px;">© 2002–2022 PrimeKey Solutio</div></div><!--EndFragment-->
</body>
</html>
```

**Reply by s0p4LiN** [reply]
![image](https://user-images.githubusercontent.com/68989865/205334571-112ceed0-7b04-4802-ab1a-a1a28b069cc6.png)

**Reply by s0p4LiN** [reply]
I did just relaunch again the Post Upgrade and it worked, I think there was some timeout during the 7.4.0 --> 7.10 in the first time and that is why It failed (I presume).

Anyway, It is upgraded ! Yeah ! 

What are the nexts steps ? Can I upgrade Wildfly ? OpenJDK ? Mariadb without issue ?

![image](https://user-images.githubusercontent.com/68989865/205398602-b935ab80-6779-4774-8b2c-04e26b9121bd.png)

**Reply by mike-agrenius-kushner** [reply]
Hi, 

Love to see it. 

Yeah, you're free to upgrade all of the above, though be aware that for the moment our JDK17-support is still a bit spotty, so I recommend sticking to JDK11 until the next major coming out in the spring. 

Cheers,
Mike

**Reply by s0p4LiN** [reply]
Tank you for you help I will upgrade it.

**Reply by s0p4LiN** [reply]
Last question:

Is it normal that it display now Working Copy ?
![image](https://user-images.githubusercontent.com/68989865/205600671-06d77681-6e9e-4264-b07b-a8b7a340f533.png)

**Comment by mike-agrenius-kushner** [comment]
Yeah, that just shows that it's a release and not a direct checkout from Git.

Cheers,
Mike

**Reply by s0p4LiN** [reply]
alright, thank you very much ! I will now write a documentation for my company before leaving and advise them to take the Enterprise Subscription for the ACME auto enroll feature...
